sw/device/tests/rstmgr_cpu_info_test.c

```diff
@@ -22,181 +21,291 @@ OTTF_DEFINE_TEST_CONFIG();
 
 /**
  * RSTMGR CPU INFO TEST
- *  This test creates a double fault by accessing a register with
- *  a non-existing address.
- *  After the double fault, the dut gets reset by the watch dog bite,
- *  and the test collects / checks the cpu_info from the rstmgr.
+ *
+ * This has three stages:
+ *
+ * 1. After the first startup, a illegal memory access is performed.
+ *    In the exception handler, a software reset is triggered.
+ *
+ * 2. After the software reset, the CPU info dump is checked against
+ *    the expected values for this single fault. The watch dog is then set up
+ *    and another illegal memory access is performed. Only this time
+ *    the exception handler performs another illegal read.
+ *    Causing the ibex to be haulted by the alert handler.
+ *    The watch dog will eventually trigger a reset.
+ *
+ * 3. After the watchdog reset, the CPU info dump is checked against
+ *    the expected values for this double fault.
  */
 
-// non existing address
-#define kIllegalAddr1 0x4041FFF0u
-#define kIllegalAddr2 0x40003618u
-#define kSkipComp 0x12345678u
 #define kCpuDumpSize 8
 
-/**
- * Cpu dump struct index
- */
+// Unmapped Addresses.
 enum {
-  kCpuDumpIdxCurrentExceptionAddr = 0,
-  kCpuDumpIdxCurrentExceptionPc = 1,
-  kCpuDumpIdxCurrentLastDataAddr = 2,
-  kCpuDumpIdxCurrentNextPc = 3,
-  kCpuDumpIdxCurrentPc = 4,
-  kCpuDumpIdxPreviousExceptionAddr = 5,
-  kCpuDumpIdxPreviousExceptionPc = 6,
-  kCpuDumpIdxPreviousValid = 7,
+  kIllegalAddr0 = 0xF0000000,
+  kIllegalAddr1 = 0xF0000001,
+  kIllegalAddr2 = 0xF0000002,
```

Comment:
It might not make a difference here, but often it's good to ensure that the separate addresses differ by at least the word size (or, conversely, also have cases where that's not the case), as that's often an edge case.

Reply:
They are only dereferenced into `uint8_t`, but I use `mmio_region_read32` so I think it's a fair point.
I have separated them into 32bit aligned addresses, also made `addr_val` a `uint32_t`.